### ISSUE TITLE
Move into upstream Homebrew repo

### ISSUE THREAD

**Issue by lwolfsonkin** [laughedelic/brew-completions#9]
In Homebrew/brew#6172, I proposed integrating completions from this repo as well as [the main fish shell repo](https://github.com/fish-shell/fish-shell/blob/master/share/completions/brew.fish) into Homebrew itself in order to guarantee wider availability and quicker updates.

Pending your agreement, I would love to combine these together and get them into Homebrew itself. Thoughts?

**Comment by laughedelic** [comment]
Hi @lwolfsonkin. Thank you for bringing it up! I completely support this proposal. I have several points that I'd like to address first:

As the readme stated from the beginning:

> This plugin is in beta, so you're welcome to try it out and report any issues you found. The code also includes some FIXME and TODO notes marking potential improvements/issues. [...]
> 
> Once it's considered stable, it may get either included with Homebrew or replace completions shipped with Fish.

I guess it can be called stable, but I would still walk through the FIXME/TODOs before pushing it to the Homebrew repo. 

As you've looked in both places, could you tell me, is there anything missing here that the completions from the main repo have?

And lastly, there is an open issue about caching #1, which I didn't have time to solve completely. I'd like to look into that to see if anything has changed or it can be closed as won't-fix.

**Comment by lwolfsonkin** [comment]
Hi @laughedelic,

> I guess it can be called stable, but I would still walk through the FIXME/TODOs before pushing it to the Homebrew repo.

I walked through some of the FIXME/TODOs in Homebrew/brew#6217 and did some cleanups of the completions, like some renames and making `brew unpin` list only non-pinned formulae.

> As you've looked in both places, could you tell me, is there anything missing here that the completions from the main repo have?

I've walked through them both a bit and added some minor additions, but I was hard pressed to find things that the fish repo's completions support and yours do not.

> And lastly, there is an open issue about caching #1, which I didn't have time to solve completely. I'd like to look into that to see if anything has changed or it can be closed as won't-fix.

For now I've punted on the caching, and I figure that can be added later. It seems that the [brew](https://github.com/Homebrew/brew/blob/master/completions/bash/brew) completions that Homebrew supports don't yet have any sort of caching, though the [zsh](https://github.com/Homebrew/brew/blob/master/completions/zsh/_brew) completions do have something. Maybe we can get inspiration from there for a caching strategy.

Lastly, the maintainers at Homebrew/brew#6217 wanted to make sure you agreed to their license, and then we're good to go to include in Homebrew. :) Thanks so much for the awesome work you put in to bring this to fruition!
